### PR TITLE
Update b tagging sf

### PR DESCRIPTION
**Pull request by jalmond**
I also added the code to make the btagging efficiency and the script to make the efficiency txt files. Not i updated the jet eta and pt range from 30, 2.5 to 20, 2.4, as this is what seemed to be used in the example from pog. This is why the efficiencies in the 20-40 bin are slightly different from previous.

### CODE REVIEW

**Comment by jalmond** [comment]
I also added function to exit btagger code and give error message if the setup is wrong.

**Comment by jedori0228** [comment]
I am not using b-tagging for now, so @choij1589 Can you confirm it's okay?

File: AnalyzerTools/src/BTagSFUtil.C [hunk 1]
```diff
@@ -190,7 +190,7 @@ float BTagSFUtil::GetJetSF(int JetFlavor, float JetPt, float JetEta) {
       double lumi_periodF = 3.157020934;
       double lumi_periodG = 7.549615806;
       double lumi_periodH = 8.545039549 + 0.216782873;
-      double total_lumi = (lumi_periodB+lumi_periodC+lumi_periodD+lumi_periodE+lumi_periodF+lumi_periodG+lumi_periodH);
+      double total_lumi =  (lumi_periodB+lumi_periodC+lumi_periodD+lumi_periodE+lumi_periodF+lumi_periodG+lumi_periodH);
```
Comment (jedori0228) [inline AnalyzerTools/src/BTagSFUtil.C]: It's less than <1 % difference, but I got


B_ver2 | 272007 | 275376 | 5750.490644035
-- | -- | -- | --
C | 275657 | 276283 | 2572.903488748
D | 276315 | 276811 | 4242.291556970
E | 276831 | 277420 | 4025.228136967
F | 277772 | 278808 | 3104.509131800
G | 278820 | 280385 | 7575.824256098
H | 280919 | 284044 | 8650.628380028

with the commands

brilcalc lumi -u /pb --normtag /cvmfs/cms-bril.cern.ch/cms-lumi-pog/Normtags/normtag_PHYSICS.json -i /afs/cern.ch/cms/CAF/CMSCOMM/COMM_DQM/certification/Collisions16/13TeV/ReReco/Final/Cert_271036-284044_13TeV_ReReco_07Aug2017_Collisions16_JSON.txt  --begin 272007 --end 275376 &> log_periodB_ver2_272007_275376.log
brilcalc lumi -u /pb --normtag /cvmfs/cms-bril.cern.ch/cms-lumi-pog/Normtags/normtag_PHYSICS.json -i /afs/cern.ch/cms/CAF/CMSCOMM/COMM_DQM/certification/Collisions16/13TeV/ReReco/Final/Cert_271036-284044_13TeV_ReReco_07Aug2017_Collisions16_JSON.txt  --begin 275657 --end 276283 &> log_periodC_275657_276283.log
brilcalc lumi -u /pb --normtag /cvmfs/cms-bril.cern.ch/cms-lumi-pog/Normtags/normtag_PHYSICS.json -i /afs/cern.ch/cms/CAF/CMSCOMM/COMM_DQM/certification/Collisions16/13TeV/ReReco/Final/Cert_271036-284044_13TeV_ReReco_07Aug2017_Collisions16_JSON.txt  --begin 276315 --end 276811 &> log_periodD_276315_276811.log
brilcalc lumi -u /pb --normtag /cvmfs/cms-bril.cern.ch/cms-lumi-pog/Normtags/normtag_PHYSICS.json -i /afs/cern.ch/cms/CAF/CMSCOMM/COMM_DQM/certification/Collisions16/13TeV/ReReco/Final/Cert_271036-284044_13TeV_ReReco_07Aug2017_Collisions16_JSON.txt  --begin 276831 --end 277420 &> log_periodE_276831_277420.log
brilcalc lumi -u /pb --normtag /cvmfs/cms-bril.cern.ch/cms-lumi-pog/Normtags/normtag_PHYSICS.json -i /afs/cern.ch/cms/CAF/CMSCOMM/COMM_DQM/certification/Collisions16/13TeV/ReReco/Final/Cert_271036-284044_13TeV_ReReco_07Aug2017_Collisions16_JSON.txt  --begin 277772 --end 278808 &> log_periodF_277772_278808.log
brilcalc lumi -u /pb --normtag /cvmfs/cms-bril.cern.ch/cms-lumi-pog/Normtags/normtag_PHYSICS.json -i /afs/cern.ch/cms/CAF/CMSCOMM/COMM_DQM/certification/Collisions16/13TeV/ReReco/Final/Cert_271036-284044_13TeV_ReReco_07Aug2017_Collisions16_JSON.txt  --begin 278820 --end 280385 &> log_periodG_278820_280385.log
brilcalc lumi -u /pb --normtag /cvmfs/cms-bril.cern.ch/cms-lumi-pog/Normtags/normtag_PHYSICS.json -i /afs/cern.ch/cms/CAF/CMSCOMM/COMM_DQM/certification/Collisions16/13TeV/ReReco/Final/Cert_271036-284044_13TeV_ReReco_07Aug2017_Collisions16_JSON.txt  --begin 280919 --end 284044 &> log_periodH_280919_284044.log

Comment (jalmond) [reply AnalyzerTools/src/BTagSFUtil.C]: It's likely the normtag is updated from when the numbers were first calculated. I will fix them and update PR

File: AnalyzerTools/src/BTagSFUtil.C [hunk 2]
```diff
@@ -200,9 +200,9 @@ float BTagSFUtil::GetJetSF(int JetFlavor, float JetPt, float JetEta) {
...
 
-      float Btag_SF_B    = GetJetSFPeriodDependant(JetFlavor, JetPt, JetEta, "2017_" +TaggerName + "_297046_299329");
-      float Btag_SF_CtoE = GetJetSFPeriodDependant(JetFlavor, JetPt, JetEta, "2017_" +TaggerName + "_297020_304671");
-      float Btag_SF_EtoF = GetJetSFPeriodDependant(JetFlavor, JetPt, JetEta, "2017_" +TaggerName + "_304671_306462");
+      float Btag_SF_B    = GetJetSFPeriodDependant(JetFlavor, JetPt, JetEta, "2017_" +TaggerName + "_297046_299329", true, TaggerName);
+      float Btag_SF_CtoE = GetJetSFPeriodDependant(JetFlavor, JetPt, JetEta, "2017_" +TaggerName + "_297020_304671", true, TaggerName);
+      float Btag_SF_EtoF = GetJetSFPeriodDependant(JetFlavor, JetPt, JetEta, "2017_" +TaggerName + "_304671_306462", true, TaggerName);
       
       /// lumi taken from https://twiki.cern.ch/twiki/bin/viewauth/CMS/PdmV2017Analysis and brilcal 
       double lumi_periodB = 4.823;
```
Comment (jedori0228) [inline AnalyzerTools/src/BTagSFUtil.C]: I know this is taken from the twiki, but with the latest golden JSON, I got

Period | Run | Run | Recored lumi
-- | -- | -- | --
B | 297046 | 299329 | 4793.969902341
C | 299368 | 302029 | 9632.746394868
D | 302030 | 303434 | 4247.792714244
E | 303824 | 304797 | 9314.581016416
F | 305040 | 306462 | 13540.062032243

with

brilcalc lumi -u /pb --normtag /cvmfs/cms-bril.cern.ch/cms-lumi-pog/Normtags/normtag_PHYSICS.json -i /afs/cern.ch/cms/CAF/CMSCOMM/COMM_DQM/certification/Collisions17/13TeV/ReReco/Cert_294927-306462_13TeV_EOY2017ReReco_Collisions17_JSON.txt  --begin 297046 --end 299329 &> log_periodB_297046_299329.log
brilcalc lumi -u /pb --normtag /cvmfs/cms-bril.cern.ch/cms-lumi-pog/Normtags/normtag_PHYSICS.json -i /afs/cern.ch/cms/CAF/CMSCOMM/COMM_DQM/certification/Collisions17/13TeV/ReReco/Cert_294927-306462_13TeV_EOY2017ReReco_Collisions17_JSON.txt  --begin 299368 --end 302029 &> log_periodC_299368_302029.log
brilcalc lumi -u /pb --normtag /cvmfs/cms-bril.cern.ch/cms-lumi-pog/Normtags/normtag_PHYSICS.json -i /afs/cern.ch/cms/CAF/CMSCOMM/COMM_DQM/certification/Collisions17/13TeV/ReReco/Cert_294927-306462_13TeV_EOY2017ReReco_Collisions17_JSON.txt  --begin 302030 --end 303434 &> log_periodD_302030_303434.log
brilcalc lumi -u /pb --normtag /cvmfs/cms-bril.cern.ch/cms-lumi-pog/Normtags/normtag_PHYSICS.json -i /afs/cern.ch/cms/CAF/CMSCOMM/COMM_DQM/certification/Collisions17/13TeV/ReReco/Cert_294927-306462_13TeV_EOY2017ReReco_Collisions17_JSON.txt  --begin 303824 --end 304797 &> log_periodE_303824_304797.log
brilcalc lumi -u /pb --normtag /cvmfs/cms-bril.cern.ch/cms-lumi-pog/Normtags/normtag_PHYSICS.json -i /afs/cern.ch/cms/CAF/CMSCOMM/COMM_DQM/certification/Collisions17/13TeV/ReReco/Cert_294927-306462_13TeV_EOY2017ReReco_Collisions17_JSON.txt  --begin 305040 --end 306462 &> log_periodF_305040_306462.log

Comment (jalmond) [reply AnalyzerTools/src/BTagSFUtil.C]: The period dependant SF splits period E. Its  C-E(up to 304671)  and E(from 304672)-F. 
Can you check the lumi for E(up to 304671)?

Comment (jedori0228) [reply AnalyzerTools/src/BTagSFUtil.C]: E(303824-304671) : 7838.041571933
E(304672-304797) : 1476.539444483

**Pull request merged**